ext/MatrixAlgebraKitGenericExt.jl

```diff
@@ -0,0 +1,230 @@
+module MatrixAlgebraKitGenericExt
+
+using MatrixAlgebraKit
+using MatrixAlgebraKit: LAPACK_SVDAlgorithm, LAPACK_EigAlgorithm, LAPACK_EighAlgorithm, LAPACK_QRIteration
+using MatrixAlgebraKit: uppertriangular!
+using MatrixAlgebraKit: @algdef, Algorithm, check_input
+using MatrixAlgebraKit: sign_safe
+using GenericLinearAlgebra
+using GenericSchur
+using LinearAlgebra: I, Diagonal
+
+function MatrixAlgebraKit.default_svd_algorithm(::Type{T}; kwargs...) where {T <: StridedMatrix{<:Union{BigFloat, Complex{BigFloat}}}}
+    return BigFloat_svd_QRIteration()
+end
+
+function MatrixAlgebraKit.svd_compact!(A::AbstractMatrix{T}, USVᴴ, alg::BigFloat_svd_QRIteration) where {T <: Union{BigFloat, Complex{BigFloat}}}
+    check_input(svd_compact!, A, USVᴴ, alg)
+    U, S, V = GenericLinearAlgebra.svd(A)
+    return U, Diagonal(S), V' # conjugation to account for difference in convention
+end
+
+function MatrixAlgebraKit.svd_full!(A::AbstractMatrix{T}, USVᴴ, alg::BigFloat_svd_QRIteration)::Tuple{Matrix{T}, Matrix{BigFloat}, Matrix{T}} where {T <: Union{BigFloat, Complex{BigFloat}}}
+    check_input(svd_full!, A, USVᴴ, alg)
+    U, S, Vᴴ = USVᴴ
+    m, n = size(A)
+    minmn = min(m, n)
+    if minmn == 0
+        MatrixAlgebraKit.one!(U)
+        MatrixAlgebraKit.zero!(S)
+        MatrixAlgebraKit.one!(Vᴴ)
+        return USVᴴ
+    end
+    S̃ = zeros(eltype(S), size(S))
+    U_compact, S_compact, V_compact = GenericLinearAlgebra.svd(A)
+    S̃[1:minmn, 1:minmn] .= Diagonal(S_compact)
+    Ũ = _gram_schmidt(U_compact)
+    Ṽ = _gram_schmidt(V_compact)
+
+    copyto!(U, Ũ)
+    copyto!(S, S̃)
```

Comment:
Would it be possible to have:

```suggestion
    U = _gram_schmidt!(U, U_compact)
    V = _gram_schmidt!(V, V_compact)
```